api/src/main/java/marquez/api/SearchResource.java

```diff
@@ -49,25 +53,23 @@ public SearchResource(@NonNull final SearchDao searchDao) {
   @GET
   @Produces(APPLICATION_JSON)
   public Response search(
-      @QueryParam("q") @NotNull String query,
+      @QueryParam("q") @NotBlank String query,
       @QueryParam("filter") @Nullable SearchFilter filter,
       @QueryParam("sort") @DefaultValue(DEFAULT_SORT) SearchSort sort,
-      @QueryParam("limit") @DefaultValue(DEFAULT_LIMIT) @Min(MIN_LIMIT) int limit) {
-    return Response.ok(
-            isQueryBlank(query)
-                ? SearchResults.EMPTY
-                : searchWithNonBlankQuery(query, filter, sort, limit))
-        .build();
-  }
-
-  private static boolean isQueryBlank(@NonNull String query) {
-    return query.trim().isEmpty();
-  }
-
-  private SearchResults searchWithNonBlankQuery(
-      String query, SearchFilter filter, SearchSort sort, int limit) {
-    final List<SearchResult> results = searchDao.search(query, filter, sort, limit);
-    return new SearchResults(results);
+      @QueryParam("limit") @DefaultValue(DEFAULT_LIMIT) @Min(MIN_LIMIT) int limit,
+      @QueryParam("namespace") @Nullable String namespace,
+      @QueryParam("before") @Valid @Pattern(regexp = YYYY_MM_DD) @Nullable String before,
+      @QueryParam("after") @Valid @Pattern(regexp = YYYY_MM_DD) @Nullable String after) {
```

Comment:
@wslulciuc Since App Wizard doesn't expose `Instant` out of the box, would it be a possibility to define a custom parameter Instant converter and register it up at the `MarquezApp.java`? Or would that be overkill for the problem we're solving here?

Reply:
For now, we'll continue to use the approach taken in our [runsAPI](https://github.com/MarquezProject/marquez/blob/main/api/src/main/java/marquez/api/RunResource.java#L115):

```java
  Response markRunAs(@NonNull RunState runState, @QueryParam("at") String atAsIso) {
    runService.markRunAs(runId, runState, Utils.toInstant(atAsIso));
    return getRun();
  }
```

But, agree a more long term solution would be to define a custom query param.